config/initializers/deprecations.rb

```diff
@@ -25,3 +25,29 @@
     )
   end
 end
+
+Foreman::Application.configure do |app|
+  config.after_initialize do
+    next if Foreman.in_rake?
+    next unless VariableLookupKey.any?
```

Comment:
Aren't smart class parameters stored in the same database model?

Reply:
In the same database table, yes, but using STI (single table inheritance). Both models `PuppetClassLookupKey` (smart class params) and `VariableLookupKey` (smart variables) inherit from the `LookupKey` class (as well as ansible variables and possibly others), which is stored on the `lookup_keys` table in the database with `type` column indicating which of the different children classes a row relates to.